### PR TITLE
fix debug output in client

### PR DESCRIPTION
**Pull request by dsikich**
The client was printing debug information about gotcha wrappers by
default.

        - update unifycr-defs so it defaults to not printing debug info
        - update unifycr-internal so that debug function prints file and
          line info
        - change unifycr.c to use debug function instead of printf

### CODE REVIEW

**Comment by nedbass** [comment]
It may be useful to be able to set the debug level at runtime with an environment variable rather than at build time.

**Comment by dsikich** [comment]
@nedbass I'm also getting  a warning about a line over 80 characters (the debug print macro) in unifier-internal.h. I was using a "\" character to continue the line before it hit 80 characters, but it looks like it is reading it as one line. Is there a better way to do that and/or get around that?

**Comment by dsikich** [comment]
@nedbass oh, yeah that would be easier. I will update with a check for the env variable at runtime instead.

**Comment by nedbass** [comment]
I often use a macro something like this, where `unifycr_debug_level` would be set in `unifycr_init()`. The "unifycr:" prefix might help distinguish these debug messages from the application's output.

```
#define DEBUG(fmt, ...)                                        \
do {                                                           \
    if (unifycr_debug_level > 0)                                       \
        printf("unifycr: %s:%d: %s: " fmt "\n",                \
               __FILE__, __LINE__, __func__, ## __VA_ARGS__);  \
} while (0)
```

**Comment by dsikich** [comment]
@nedbass I'm not really sure about the warnings I am getting. When I continue a string on another line (to not go over 80 characters) it still says it is over 80 characters. If I use a \ character it also warns about that? I'm not sure. I've tried many variations, but I'm not seeing how to get the warning to go away.

**Comment by dsikich** [comment]
@nedbass okay, tests are passing. I had to fix my ~/.vimrc config.

File: client/src/unifycr.c [hunk 1]
```diff
@@ -2770,7 +2789,7 @@ chunk_list_t *unifycr_get_chunk_list(char *path)
     return NULL;
 }
 
-/* debug function to print list of chunks constituting a file
+/* DEBUG function to print list of chunks constituting a file
```
Comment (nedbass) [inline client/src/unifycr.c]: same here

File: client/src/unifycr.c [hunk 2]
```diff
@@ -2218,7 +2236,7 @@ static int unifycr_sync_to_del()
     memcpy(cmd_buf + 2 * sizeof(int),
            &local_rank_idx, sizeof(int));
     memcpy(cmd_buf + 3 * sizeof(int),
-           &dbg_rank, sizeof(int)); /*add debug info*/
+           &dbg_rank, sizeof(int)); /*add DEBUG info*/
```
Comment (nedbass) [inline client/src/unifycr.c]: I suspect this was an inadvertent search and replace hit (not that it hurts anything :)

File: client/src/unifycr-defs.h [hunk 3]
```diff
@@ -1,8 +1,15 @@
+/* holds debug level for unifycr, this defaults to zero
+ * in unifycr_init, but can be changed at runtime by setting
+ * the DEBUG env variable to a value greater than 1
```
Comment (nedbass) [inline client/src/unifycr-defs.h]: I'd prefer to name it `UNIFYCR_DEBUG` for consistency with other unifycr environment variables and  to reduce likelihood of name collision.

**Comment by dsikich** [comment]
@nedbass Thanks for your help. No worries, this will be very useful!

File: client/src/unifycr.c [hunk 4]
```diff
@@ -1679,21 +1686,30 @@ static int unifycr_abtoull(char *str, unsigned long long *val)
 
 static int unifycr_init(int rank)
 {
-    if (! unifycr_initialized) {
+    if (!unifycr_initialized) {
+
+		/* unifycr DEBUG level default is zero */
+        unifycr_debug_level = 0;
```
Comment (nedbass) [inline client/src/unifycr.c]: Indentation looks off here. Probably from mixing tabs and spaces.

I use this in my .vimrc to make that easier to spot:

```
set list            " * Turn on printing of invisible chars
set listchars=tab:>-,trail:! " * Customize the characters used to print the
                    "   invisible chars
```

File: client/src/unifycr.c [hunk 5]
```diff
@@ -1679,21 +1686,30 @@ static int unifycr_abtoull(char *str, unsigned long long *val)
...
 
         int i;
+
         for (i = 0; i < GOTCHA_NFUNCS; i++) {
             if (*(void **)(wrap_unifycr_list[i].function_address_pointer) == 0) {
-                printf("This function name failed to be wrapped: %s\n",
-                       wrap_unifycr_list[i].name);
+                DEBUG("This function name failed to be wrapped: %s\n",
+                      wrap_unifycr_list[i].name
+                      );
```
Comment (nedbass) [inline client/src/unifycr.c]: (style) move closing paren to line above

File: client/src/unifycr.c [hunk 6]
```diff
@@ -2138,19 +2154,21 @@ int unifycrfs_mount(const char prefix[], size_t size, int rank)
...
+            DEBUG("rank:%d, fail to init shared receive memory.", dbg_rank);
             return UNIFYCR_FAILURE;
         }
 
         rc = unifycr_sync_to_del();
         if (rc < 0) {
-            debug("rank:%d, fail to convey information to the delegator.", dbg_rank);
+            DEBUG("rank:%d, fail to convey information to the delegator.",
+				  dbg_rank
+				  );
```
Comment (nedbass) [inline client/src/unifycr.c]: (style) move closing paren to line above

File: client/src/unifycr.c [hunk 7]
```diff
@@ -665,7 +668,9 @@ int unifycr_fid_create_file(const char *path)
      * and return appropriate error if it is greater
      */
     strcpy((void *)&unifycr_filelist[fid].filename, path);
-    debug("Filename %s got unifycr fd %d\n", unifycr_filelist[fid].filename, fid);
+    DEBUG("Filename %s got unifycr fd %d\n",
+		  unifycr_filelist[fid].filename, fid
+		  );
```
Comment (nedbass) [inline client/src/unifycr.c]: (style) move closing paren to line above

File: client/src/unifycr.c [hunk 8]
```diff
@@ -1148,23 +1153,24 @@ int unifycr_fid_open(const char *path, int flags, mode_t mode, int *outfid,
...
-            debug("unifycr_superblock = %p; free_fid_stack = %p; free_chunk_stack = %p; unifycr_filelist = %p; chunks = %p\n",
-                  unifycr_superblock, free_fid_stack, free_chunk_stack, unifycr_filelist,
-                  unifycr_chunks
-                 );
+            DEBUG("Couldn't find entry for %s in UNIFYCR\n", path);
+            DEBUG("unifycr_superblock = %p; free_fid_stack = %p;"
+                  "free_chunk_stack = %p; unifycr_filelist = %p; chunks = %p\n",
+                  unifycr_superblock, free_fid_stack, free_chunk_stack,
+                  unifycr_filelist, unifycr_chunks
+                  );
```
Comment (nedbass) [inline client/src/unifycr.c]: (style) move closing paren to line above

File: client/src/unifycr.c (hunk 4 above)
Comment (dsikich) [reply client/src/unifycr.c]: @nedbass yes, thanks. I can see the tabs now!

File: client/src/unifycr.c [hunk 9]
```diff
@@ -2239,8 +2244,9 @@ static int unifycr_sync_to_del()
     memcpy(cmd_buf + 7 * sizeof(int) + 6 * sizeof(long),
            &data_size, sizeof(long));
 
+	/*adjust to add debug info*/
```
Comment (nedbass) [inline client/src/unifycr.c]: check indentation

**Comment by dsikich** [comment]
@nedbass the newly merged pull requests conflict with a file now. I will submit a new PR off of an updated dev branch.

**Comment by nedbass** [comment]
> @nedbass the newly merged pull requests conflict with a file now. I will submit a new PR off of an updated dev branch.

Sounds good. Just FYI, you can always force update your branch after a rebase, it's not necessary to open a new PR.

**Comment by dsikich** [comment]
Okay, thanks good to know. In this case though I’ve already submitted a new one (#82).

**Pull request closed without merging**